src/types/custom-interfaces.ts

```diff
@@ -0,0 +1,64 @@
+import { AnyToken } from './signable-token';
+import { FeeEntry } from '../api';
+
+// These custom interfaces are used because API.SignableToken is not yet a union type due to OAS 2.0 spec not supporting `OneOf`
+// As well as the fact that SignableToken has an extra `data` object which we would like to flatten
+// SignableToken is replaced with AnyToken
+
+export interface UnsignedOrderRequest {
+  buy: {
+    /**
+     * Fee-exclusive amount to buy the asset
+     * @type {string}
+     */
+    amount: string;
+    token: AnyToken;
+  };
+  sell: {
+    /**
+     * Amount to sell (quantity)
+     * @type {string}
+     */
+    amount: string;
+    token: AnyToken;
```

Comment:
This validation currently exists in the order api. 

```go
// Assert this is an order
if signableData.SellAsset.IsFungible() && signableData.BuyAsset.IsNonFungible() {
	return imxerrors.NewUser("invalid buy and sell asset type for an order")
}

```
Should we update sell token to only accept ETH/ERC20 ? as well as Buy token to `token: ERC721Token`
```suggestion
    token: ETHToken | ERC20Token;
```

@Kastet

Reply:
Update: Trading team says the above restriction will be removed in the future so I guess we can keep this code as is.